src/battle_interface.c

```diff
@@ -3470,7 +3470,10 @@ void TryAddLastUsedBallItemSprites(void)
         // we're out of the last used ball, so just set it to the first ball in the bag
         // we have to compact the bag first bc it is typically only compacted when you open it
         CompactItemsInBagPocket(&gBagPockets[BALLS_POCKET]);
-        gBallToDisplay = gBagPockets[BALLS_POCKET].itemSlots[0].itemId;
+
+        u16 firstBall = gBagPockets[BALLS_POCKET].itemSlots[0].itemId;
+        if (firstBall > ITEM_NONE)
+            gBallToDisplay = gBagPockets[BALLS_POCKET].itemSlots[0].itemId;
```

Comment:
Please move `u16 firstBall = gBagPockets[BALLS_POCKET].itemSlots[0].itemId;` to the top of the scope *(in other words before the comment a few lines above with an empty space inbetween)*, because this won't build on agbcc.